### DISCUSSION TITLE
Official support for client libraries

### DISCUSSION THREAD

**Discussion by thecoolwinter** [supabase/supabase#2650] [Questions] [answered]
There are [loads of client libraries](https://github.com/supabase/supabase/discussions/5) out there that are being actively developed, are there set parameters for becoming officiall clients? Eg: JS has first class support, but swift and rust do not, is there a path for their libraries to be an official client?

**Comment by silentworks** [comment, marked as answer]
Over time more libraries will become official, but also some of these libraries can stay independent as sometimes the creators of the libraries prefer it to stay that way. Supabase has an official Swift library from what I can tell, it might not be as feature-complete as the JavaScript one but this is because first priority was to get the JavaScript client library ready first.
